code/modules/mob/living/simple_animal/hostile/cockroach.dm

```diff
@@ -106,3 +106,107 @@
+	icon_state = "glockroach-umbra"
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	obj_damage = 30
+	gold_core_spawnable = NO_SPAWN
+	projectiletype = /obj/projectile/glockroachbullet/umbra
+	casingtype = /obj/item/ammo_casing/glockroach/umbra
+
+/obj/projectile/glockroachbullet/umbra
+	damage = 20 //double damage
+
+/obj/item/ammo_casing/glockroach/umbra
+	name = ".045 ACP bullet casing"
+	desc = "This is cursed on so many levels."
+	projectile_type = /obj/projectile/glockroachbullet/umbra
+
+/mob/living/simple_animal/hostile/cockroach/glockroach/magnum
+	name = "magnumroach"
+	desc = "Three five rot- Revolveroa- MAGNUMROACH." //reference to hiimdaisy
+	icon_state = "magnumroach"
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	obj_damage = 40
+	gold_core_spawnable = NO_SPAWN
+	projectilesound = 'sound/weapons/gun/revolver/shot.ogg'
+	projectiletype = /obj/projectile/glockroachbullet/magnum
+	casingtype = /obj/item/ammo_casing/glockroach/magnum
+	health = 100 //The miniboss.
+	maxHealth = 100
+
+/mob/living/simple_animal/hostile/cockroach/glockroach/magnum/make_squashable()
+	AddElement(/datum/element/squashable, squash_chance = 50, squash_damage = 100) //they're a glass cannon
+
+/obj/projectile/glockroachbullet/magnum
+	damage = 60 //its a 357
+
+/obj/item/ammo_casing/glockroach/magnum
+	name = ".0357 bullet casing"
+	desc = "Itty bitty revolver ammunition."
+	projectile_type = /obj/projectile/glockroachbullet/magnum
+
+/mob/living/simple_animal/hostile/cockroach/glockroach/deagle
+	name = "deagleroach"
+	desc = "The king of the glockroaches. Mutations have caused them to have a desert eagle."
+	icon_state = "deagleroach"
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	obj_damage = 40 //double damage
+	gold_core_spawnable = NO_SPAWN
+	projectiletype = /obj/projectile/glockroachbullet/deagle
+	casingtype = /obj/item/ammo_casing/glockroach/deagle
+	squish_chance = 1 //1 percent chance. Take your best shot.
+	health = 200 //hardy
+	maxHealth = 200
+
+/mob/living/simple_animal/hostile/cockroach/glockroach/deagle/make_squashable()
+	AddElement(/datum/element/squashable, squash_chance = 1, squash_damage = 200)
+
+/obj/projectile/glockroachbullet/deagle
+	damage = 75 //deagle is stronk.
+
+/obj/item/ammo_casing/glockroach/deagle
+	name = ".05 AE bullet casing"
+	desc = "A tiny desert eagle bullet casing."
+	projectile_type = /obj/projectile/glockroachbullet/deagle
+
+/mob/living/simple_animal/hostile/cockroach/glockroach/god
+	name = "godroach"
+	desc = "Life...Hope...Dreams...where do they come from? And where are they headed? All these things I AM GOING TO DESTROY!"
+	icon_state = "godroach"
+	melee_damage_lower = 9001
+	melee_damage_upper = 9001
+	obj_damage = 9001
+	gold_core_spawnable = NO_SPAWN
+	projectiletype = /obj/projectile/glockroachbullet/god
+	casingtype = /obj/item/ammo_casing/glockroach/god
+	squish_chance = 0 //oh no
+	health = 13372462 //oh NO
+	maxHealth = 13372462
+	ranged_cooldown_time = 1
```

Comment:
Did you want the glockroach to fire once every tick?

Reply:
Yes.